torch/_inductor/lowering.py

```diff
@@ -258,41 +258,73 @@ def in_namespace(op, namespace):
     return False
 
 
-def transform_args(args, broadcast, type_promotion_kind, convert_input_to_bool):
-    indices = [i for i, x in enumerate(args) if isinstance(x, TensorBox)]
-    if (type_promotion_kind or convert_input_to_bool) and indices:
+def transform_args(
+    args: List[Any],
+    kwargs: Dict[str, Any],
+    broadcast: bool,
+    type_promotion_kind: Optional[ELEMENTWISE_TYPE_PROMOTION_KIND],
+    convert_input_to_bool: bool,
+) -> Tuple[List[Any], Dict[str, Any]]:
+    args_indices = [i for i, x in enumerate(args) if isinstance(x, TensorBox)]
+    kwargs_indices = [k for k, v in kwargs.items() if isinstance(v, TensorBox)]
+    # check that there's something to transform
+    if not args_indices and not kwargs_indices:
+        return args, kwargs
+
+    if type_promotion_kind or convert_input_to_bool:
         if convert_input_to_bool:
             dtype = torch.bool
         else:
             # FIXME that's a crude approximation for promoting args
             promoting_args = [
                 a
-                for a in args
-                if isinstance(a, (Number, sympy.Basic))
-                or getattr(a, "dtype", None) is not None
+                for a in itertools.chain(args, kwargs.values())
+                if isinstance(a, (Number, sympy.Basic)) or hasattr(a, "dtype")
```

Comment:
What do you think ? For args, keep logic as it was. For kwargs, only include Tensors in promoting dtypes. 

I know this is a little crude, but see fixme below. And the failures from the lowering of amax make me a little worried. Im not sure we have sufficient opinfo coverage of promoting types for each opinfo to validate the changes here so would like to be conservative